cmake/CPM.cmake

```
@@ -46,6 +46,13 @@ set(CURRENT_CPM_VERSION 1.0.0-development-version)
 
 get_filename_component(CPM_CURRENT_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
 if(CPM_DIRECTORY)
+  if(${CMAKE_VERSION} VERSION_LESS "3.17.0")
+    get_directory_property(hasParent PARENT_DIRECTORY)
+    if(NOT CPM_DIRECTORY STREQUAL CPM_CURRENT_DIRECTORY OR ${hasParent})
```

Additionally the test cases seem to fail if `hasParent` is an empty string.

```
1: CMake Error at /home/runner/work/CPM.cmake/CPM.cmake/cmake/CPM.cmake:51 (if):
1:   if given arguments:
1: 
1:     "NOT" "CPM_DIRECTORY" "STREQUAL" "CPM_CURRENT_DIRECTORY" "OR"
1: 
1:   Unknown arguments specified
1: Call Stack (most recent call first):
```

It should be fixable by adding quotation marks.

```suggestion
    if(NOT CPM_DIRECTORY STREQUAL CPM_CURRENT_DIRECTORY OR "${hasParent}")
```